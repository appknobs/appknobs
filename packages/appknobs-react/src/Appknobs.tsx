import * as React from 'react'
import {AppknobsWrapperLogic} from './AppknobsWrapperLogic'
import {Context} from './Context'

export class Appknobs extends AppknobsWrapperLogic {
  public render() {
    const {features, client, payload, ...rest} = this.props

    return (
      <Context.Provider value={this.state.features}>
        {React.cloneElement(
          this.props.children as React.ReactElement<any>,
          rest,
        )}
      </Context.Provider>
    )
  }
}
