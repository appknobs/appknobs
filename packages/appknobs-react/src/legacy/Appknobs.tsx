import * as propTypes from 'prop-types'
import * as React from 'react'
import {AppknobsWrapperLogic} from '../AppknobsWrapperLogic'

export class Appknobs extends AppknobsWrapperLogic {
  public getChildContext() {
    return {
      features: this.state.features,
    }
  }

  public render() {
    const {children, features, client, payload, ...rest} = this.props

    return React.cloneElement(children as React.ReactElement<any>, rest)
  }
}

// @ts-ignore
Appknobs.childContextTypes = {
  features: propTypes.object,
}
