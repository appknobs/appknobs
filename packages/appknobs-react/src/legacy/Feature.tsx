import * as propTypes from 'prop-types'
import {Features} from '../AppknobsWrapperLogic'
import {FeatureProps, renderFeature} from '../renderFeature'

interface Context {
  features: Features
}

export const Feature = (props: FeatureProps, {features}: Context) => {
  return renderFeature(props, features)
}

// @ts-ignore
Feature.contextTypes = {
  features: propTypes.any,
}
