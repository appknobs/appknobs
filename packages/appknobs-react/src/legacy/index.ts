export {Appknobs} from './Appknobs'
export {Feature} from './Feature'
