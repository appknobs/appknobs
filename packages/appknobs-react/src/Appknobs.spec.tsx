import * as enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import * as React from 'react'
import * as testRenderer from 'react-test-renderer'
import {Appknobs as AppknobsCurrent} from './Appknobs'
import {Appknobs as AppknobsLegacy} from './legacy/Appknobs'

enzyme.configure({adapter: new Adapter()})

const runTest = (version, Appknobs) => {
  describe(`Appknobs ${version}`, () => {
    it('renders children', async () => {
      const Comp = () => <span>12</span>

      const tree = testRenderer
        .create(
          <Appknobs client={null}>
            <Comp />
          </Appknobs>,
        )
        .toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
}

runTest('current', AppknobsCurrent)
runTest('legacy', AppknobsLegacy)
