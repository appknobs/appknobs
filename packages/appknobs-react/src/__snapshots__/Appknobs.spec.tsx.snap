// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appknobs current renders children 1`] = `
<span>
  12
</span>
`;

exports[`Appknobs legacy renders children 1`] = `
<span>
  12
</span>
`;
